src/tests/ftest/util/environment_utils.py

```diff
@@ -0,0 +1,560 @@
+
+    @staticmethod
+    def default_insecure_mode():
+        """Get the default insecure mode.
+
+        Returns:
+            str: the default insecure mode
+        """
+        return "True"
+
+    @property
+    def bullseye_src(self):
+        """Get the bullseye source file.
+
+        Returns:
+            str: the bullseye source file
+        """
+        return os.environ.get(self.__ENV_VAR_MAP['bullseye_src'])
+
+    @bullseye_src.setter
+    def bullseye_src(self, value):
+        """Set the bullseye source file.
+
+        Args:
+            value (str, bool): the bullseye source file
+        """
+        self.__set_value('bullseye_src', value)
+
+    @staticmethod
+    def default_bullseye_src():
+        """Get the default bullseye source file.
+
+        Returns:
+            str: the default bullseye source file
+        """
+        return os.path.join(os.path.dirname(os.path.abspath(__file__)), "test.cov")
+
+    @property
+    def bullseye_file(self):
+        """Get the bullseye file.
+
+        Returns:
+            str: the bullseye file
+        """
+        return os.environ.get(self.__ENV_VAR_MAP['bullseye_file'])
+
+    @bullseye_file.setter
+    def bullseye_file(self, value):
+        """Set the bullseye file.
+
+        Args:
+            value (str, bool): the bullseye file
+        """
+        self.__set_value('bullseye_file', value)
+
+    @staticmethod
+    def default_bullseye_file():
+        """Get the default bullseye file.
+
+        Returns:
+            str: the default bullseye file
+        """
+        return os.path.join(os.sep, "tmp", "test.cov")
+
+
+def set_test_environment(logger, build_vars_file, test_env=None, servers=None, clients=None,
+                         provider=None, insecure_mode=False, details=None):
+    """Set up the test environment.
+
+    Args:
+        logger (Logger): logger for the messages produced by this method
+        build_vars_file (str): the full path to the DAOS build_vars.json file
+        test_env (TestEnvironment, optional): the current test environment. Defaults to None.
+        servers (NodeSet, optional): hosts designated for the server role in testing. Defaults to
+            None.
+        clients (NodeSet, optional): hosts designated for the client role in testing. Defaults to
+            None.
+        provider (str, optional): provider to use in testing. Defaults to None.
+        insecure_mode (bool, optional): whether or not to run tests in insecure mode. Defaults to
+            False.
+        details (dict, optional): dictionary to update with interface and provider settings if
+            provided. Defaults to None.
+
+    Raises:
+        TestEnvironmentException: if there is a problem setting up the test environment
+
+    """
+    logger.debug("-" * 80)
+    logger.debug("Setting up the test environment variables")
+
+    if test_env:
+        # Update the PATH environment variable
+        update_path(logger, build_vars_file)
```

Comment:
build_vars_file is required, but test_env is not. But build_vars_file is only used when test_env is passed?

Reply:
Fixed. thanks.